src/librustdoc/html/layout/tests.rs

```diff
@@ -0,0 +1,24 @@
+#[test]
```

Comment:
You can put this unit test directly at the bottom of `layout.rs` inside:

```rust
#[cfg(test)]
mod tests {
   // your code
}
```

Since there is only one test, I don't think it's worth creating a whole new file.